server/models/Collective.js

```diff
@@ -1560,21 +1569,55 @@ export default function (Sequelize, DataTypes) {
     return this;
   };
 
+  /**
+   * Update the currency of a "Collective" row (account)
+   *
+   * This is a safe version that can only be used by Users and Organizations that are not hosts
+   */
   Collective.prototype.updateCurrency = async function (currency, remoteUser) {
     if (typeof currency === undefined || !remoteUser || currency === this.currency || !remoteUser.isAdmin(this.id)) {
       return;
     }
+
     const error = 'Only Users and Organisation that are not hosts can update currency';
-    if (this.type === types.USER || this.type === types.ORGANIZATION) {
-      const isHost = await this.isHost();
-      if (isHost) {
-        throw new Error(error);
-      }
-      this.update({ currency });
-    } else {
+    if (![types.USER, types.ORGANIZATION].includes(this.type)) {
       throw new Error(error);
     }
-    return this;
+    const isHost = await this.isHost();
+    if (isHost) {
+      throw new Error(error);
+    }
+
+    return this.setCurrency(currency);
+  };
+
+  /**
+   * Set the currency of a "Collective" row (account)
+   *
+   * This is meant to be used internally, no access control.
+   */
+  Collective.prototype.setCurrency = async function (currency) {
+    const isHost = await this.isHost();
+    if (isHost) {
+      // We only expect currency change at the beginning of the history of the Host
+      const transactionCount = await models.Transaction.count({ where: { HostCollectiveId: this.id } });
+      if (transactionCount > 0) {
+        throw new Error(
+          'You cannot change the currency of an Host with transactions. Please contact support@opencollective.com.',
+        );
+      }
+      const collectives = await this.getHostedCollectives();
+      // We use setCurrency so that it will cascade to Tiers
+      await Promise.map(collectives, collective => collective.setCurrency(currency), { concurrency: 3 });
+    }
+
+    // This is currently for COLLECTIVE and EVENTS but we make it generic
+    const tiers = await this.getTiers();
+    if (tiers.length > 0) {
+      await Promise.map(tiers, tier => tier.update({ currency }), { concurrency: 3 });
```

Comment:
Should we recalculate the tier value based on FXRate?
I understand why this wouldn't be necessary for some currencies, but there are currencies that this update wouldn't make sense, for example, USD to Yen.

Reply:
I thought about this problem but I don't want to go that way and end up with weird values.

We could delete the tiers completely. But, if it's the default tiers, we're not smart enough anyway, it's what we're currently doing (same for all currencies):

```
  if (tiers.length === 0) {
    tiers.push({
      type: 'TIER',
      name: 'backer',
      slug: 'backers',
      amount: 500,
      presets: [500, 1000, 2500, 5000],
      interval: 'month',
      currency: currency,
      minimumAmount: 500,
      amountType: 'FLEXIBLE',
    });
    tiers.push({
      type: 'TIER',
      name: 'sponsor',
      slug: 'sponsors',
      amount: 10000,
      presets: [10000, 25000, 50000],
      interval: 'month',
      currency: currency,
      minimumAmount: 10000,
      amountType: 'FLEXIBLE',
    });
  }
```

If we're in this situation, we're really at the beginning of the collective existence, we don't expect to have processed any transaction.

I don't see a simple solution so we will keep it like that for now.

Reply:
Agreed, I don't think there's an easy solution for this. :shipit: